src/main/java/com/vmware/vipclient/i18n/messages/api/opt/server/RemoteLocaleOpt.java

```diff
@@ -30,61 +27,49 @@ public RemoteLocaleOpt(LocaleDTO dto) {
         this.dto = dto;
     }
 
-    public Map<String, String> getRegions(String locale) {
+    public void getRegions(String locale, LocaleCacheItem cacheItem) {
         logger.debug("Look for regions from Singleton Service for locale [{}]!", locale);
-    	Map<String, Object> response = VIPCfg.getInstance().getVipService().getHttpRequester().request(
-                V2URL.getRegionListURL(locale, VIPCfg.getInstance().getVipService().getHttpRequester().getBaseURL()),
-                ConstantsKeys.GET, null);
-    	String responseData = (String) response.get(URLUtils.BODY);
-        if(responseData == null || responseData.isEmpty())
-            return null;
-        Map<String, String> respMap = null;
         try {
-            JSONObject jsonObject = (JSONObject) JSONValue.parseWithException(responseData);
-            JSONArray jsonArray = (JSONArray) jsonObject.get(ConstantsKeys.DATA);
-            if (jsonArray != null && !jsonArray.isEmpty()) {
-                Map<String, Object> regionMap = JSONUtils.getMapFromJson(jsonArray.get(0).toString());
-                respMap = (Map<String, String>) regionMap.get(ConstantsKeys.TERRITORIES);
+            List<Map<String, Object>> dataNode = (List<Map<String, Object>>) getDataFromResponse(
+                V2URL.getRegionListURL(locale, VIPCfg.getInstance().getVipService().getHttpRequester().getBaseURL()),
+                ConstantsKeys.GET, null, cacheItem);
+            if (dataNode != null && !dataNode.isEmpty()) {
+                Map<String, Object> regionMap = dataNode.get(0);
+                Map<String, String> territories = (Map<String, String>) regionMap.get(ConstantsKeys.TERRITORIES);
+                if (territories != null) {
+                    cacheItem.addCachedData(territories);
```

Comment:
Set timestamp, maxAgeMillis, etag and cachedData using 1 single synchronized setter for thread-safety. Please see my comment in FormatCacheItem. See recent changes in ComponentBasedOpt.getComponentMessages for reference.

Suggestion: Because you have 2 child classes RemoteLocaleOpt and RemotePatternOpt, call the new synchronized setter method that sets all 4 (timestamp, maxAgeMillis, etag and cachedData) in these child classes. Do not pass the original cacheItem object to getDataFromResponse anymore. Instead, inside getDataFromResponse, just return a map that holds the timestamp, etag, maxAgeMillis and responseBody object from the response from the service. Then in here (the child class), use the values in the  map to populate the original cacheItem object using the single synchronized setter method I mentioned above.